src/HttpHeader.cc

```diff
@@ -759,6 +773,16 @@ HttpHeader::addEntry(HttpHeaderEntry * e)
 
     debugs(55, 7, this << " adding entry: " << e->id << " at " << entries.size());
 
+    // verify whether adding the entry will not exceed the maximum header size allowed
+    const auto max = maxLen();
+    if (max != UnrestrictedSize) {
+        const auto eLen = e->length();
```

Comment:
Please move eLen declaration outside so that we do not have to compute the entry length twice. This is not an efficiency/performance concern. I just want to avoid an implication that the entry length may change between the check and the `len` increment.

Also some minor renaming, mostly for consistency sake. Please propagate as needed, of course.

```suggestion
    const auto maxLen = maxPackedSize();
    const auto eLen = e->length();
    if (maxLen != UnrestrictedSize) {
```

Reply:
I removed this block after addressing another change [request](https://measurementfactory.atlassian.net/browse/SQUID-479?focusedCommentId=43555&oldIssueView=true&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-43555).